src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/ResourceScope.java

```diff
@@ -191,10 +191,11 @@
 
     /**
      * Add a custom cleanup action which will be executed when the resource scope is closed.
+     * The order in which custom cleanup actions are invoked once the scope is closed is unspecified.
      * @param runnable the custom cleanup action to be associated with this scope.
      * @throws IllegalStateException if this scope has already been closed.
      */
-    void addOnClose(Runnable runnable);
+    void addCloseAction(Runnable runnable);
```

Comment:
The rename to addCloseAction is good, it better describes what the methods does and ties in nicely with "cleanup action".